normandy/control/static/control/js/actions/ControlActions.js

```diff
@@ -183,26 +148,33 @@ function setNotification(notification) {
 function makeApiRequest(requestType, settings) {
   return (dispatch, getState) => {
     let apiRequestConfig = apiRequestMap[requestType](settings, getState);
-    if (apiRequestConfig) {
-      dispatch(requestInProgress());
-      return apiFetch(apiRequestConfig.url, {
-        ...API_REQUEST_SETTINGS,
-        ...apiRequestConfig.settings
-      })
-      .then(response => {
-        dispatch(requestComplete({ status: 'success', notification: apiRequestConfig.successNotification }));
-        dispatch(apiRequestConfig.actionOnSuccess(apiRequestConfig.successActionParams || response));
-      })
-      .catch(error => {
+
+    dispatch(requestInProgress());
+
+    return fetch(apiRequestConfig.url, {
+      ...API_REQUEST_SETTINGS,
+      ...apiRequestConfig.settings
+    })
+    .then(response => {
+      if (response.status >= 400) {
         dispatch(requestComplete({ status: 'error', notification: apiRequestConfig.errorNotification }));
-      });
-    }
+        return response.json().then(err => { throw err; });
```

Comment:
Does the error need to be thrown here? I'd think that the promise chain would surface the error to the code calling the API automatically.